### ISSUE TITLE
1.0.0-beta.1 version has low accuracy

### ISSUE THREAD

**Issue by haozes** [tryolabs/TLSphinx#40]
I write a demo with 1.0.0-beta.1 version. It has low accuracy.
Is it sample rate config problem?  

**Decorder.swift** set input 32bit,44.1khz ,output 16bit，44.1khz
`

        let formatIn = AVAudioFormat(commonFormat: .pcmFormatFloat32, sampleRate: 44100, channels: 1, interleaved: false)            

        let formatOut = AVAudioFormat(commonFormat: .pcmFormatInt16, sampleRate: 44100, channels: 1, interleaved: false)        
`

And in your unittest file:**basic.swift**  testSpeechFromFile() method, **goforward.raw** is 8K,32bit ，but it pass unittest!  why?

**Comment by BrunoBerisso** [comment]
Hi.

If you are using `decodeSpeechAtPath` in your demo you need to make sure the input format match the format you pass in to the `Config`. The input and output formats you past here are not the one used to decode the speech in a file.

It you tell me more about your test maybe I can help you improve your results :)

**Comment by haozes** [comment]
Thanks for your reply.
I test with iPhone7 use **startDecodingSpeech** with live stream.   I have same issue as :  

> https://github.com/tryolabs/TLSphinx/issues/38

with your origin code:



         let formatIn = AVAudioFormat(commonFormat: .pcmFormatFloat32, sampleRate: 16000, channels: 1, interleaved: false)    
 
         let formatOut = AVAudioFormat(commonFormat: .pcmFormatInt16, sampleRate: 16000, channels: 1, interleaved: false)



**speechstate** always silence, so I change  to :


        let formatIn = AVAudioFormat(commonFormat: .pcmFormatFloat32, sampleRate: 44100, channels: 1, interleaved: false)     

        let formatOut = AVAudioFormat(commonFormat: .pcmFormatInt16, sampleRate: 44100, channels: 1, interleaved: false)


it works ,but the result is very low accuracy. And I also tried 32KHZ,48KHZ, which sample rate config is right?

**Comment by haozes** [comment]
I think i got the problem
pocketsphinx needs 16KHZ samples. the iPhone7/iOS10 default record audio format is 44.1K 2 channels sample.

The problem is when i set input format to 44.1KHZ, and convert to 16KHZ use AVAudioConverter failed with error:

> Error Domain=NSOSStatusErrorDomain Code=-50 "(null)"
> Error Domain=NSOSStatusErrorDomain Code=-50 "(null)"
> Error Domain=NSOSStatusErrorDomain Code=-50 "(null)"

My code:
`

       let formatIn =  input.inputFormat(forBus: 0)

        //let formatIn = AVAudioFormat(commonFormat: .pcmFormatFloat32, sampleRate: 44100, channels: 2, interleaved: false)

        let formatOut = AVAudioFormat(commonFormat: .pcmFormatInt16, sampleRate: 16000, channels: 1, interleaved: false)
        let bufferMapper = AVAudioConverter(from: formatIn, to: formatOut)

        mixer.installTap(onBus: 0, bufferSize: 4096, format: formatIn, block: {
            [unowned self] (buffer: AVAudioPCMBuffer!, time: AVAudioTime!) in

            let sphinxBuffer = AVAudioPCMBuffer(pcmFormat: formatOut, frameCapacity: buffer.frameCapacity)

            //This is needed because the 'frameLenght' default value is 0 (since iOS 10) and cause the 'convert' call
            //to faile with an error (Error Domain=NSOSStatusErrorDomain Code=-50 "(null)")
            //More here: http://stackoverflow.com/questions/39714244/avaudioconverter-is-broken-in-ios-10
            sphinxBuffer.frameLength = sphinxBuffer.frameCapacity

            do {
                try bufferMapper.convert(to: sphinxBuffer, from: buffer)
                
            } catch(let error as NSError) {
                print(error)
                return
            }

`
  **try bufferMapper.convert(to: sphinxBuffer, from: buffer) ,output error here.**

But I still don't know how to fix it.

**Comment by BrunoBerisso** [comment]
Awesome, do you test on a real device?

The accuracy can be related to the model you are using. Keep in mind that if you want to use CMUSphinx for something like dictation you need to narrow the LM to some domain to get near to acceptable results. Using a general LM will not work as expected.

**Comment by haozes** [comment]
Yes, I test it on real device. I use it to keyword spotting(Chinese Language),the model is download form pocketsphinx package. 

PS: TLSphinx only output result after speech stop ,  output keywords immediately when  recognize keyword well be useful

**Comment by BrunoBerisso** [comment]
Do you mind [open a PR](https://github.com/tryolabs/TLSphinx/compare/development?expand=1) with your changes? I think you may fix #24 witch is a long standing issue 😄  I also add #41 to try to get an hypothesis before the end of speech. I can't get on this now but maybe someone else can.

About the low accuracy, how are you performing the test? do you try the same test with `pocketsphinx_continuous`? (more on this [here](https://cmusphinx.github.io/wiki/tutorialpocketsphinx/))
